gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/GoogleHadoopFSInputStream.java

```diff
@@ -98,6 +112,17 @@ public synchronized int read() throws IOException {
     return (b & 0xff);
   }
 
+  private void captureDuration(String suffix, Stopwatch stopwatch, int byteRead) {
+    if (isTraceLoggingEnabled) {
+      Map<String, Object> jsonMap = new HashMap<>();
+      // Capturing time in nanos because majority of the reads will be done from cache
+      long nanos = stopwatch.elapsed(TimeUnit.NANOSECONDS);
```

Comment:
What is the value in capturing this at the nano second granularity. It might be better to keep this consistent with other places where we are using milli seconds.

Reply:
> Capturing time in nanos because majority of the reads will be done from cache

As it is being read from in-memory cache it will not be captured as part of millisec.

Reply:
I doubt we care much about the local cache access latencies. It would be better to keep this consistent with other places and keep this in MS.

Reply:
We do care for it if calls are in magnitude 10^7. As far as other places i.e. calls to gcs, is concerned it's always in millisecond magnitude. So, we can bypass the nanosec precision but same argument can't be made for connector level APIs, specifically `read`.